### PR TITLE
[ONNX] More debug logging from fx to onnx

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107654
* #107653
* #107409
* #107408

Summary:
- Log fx graph name for 'fx-graph-to-onnx' diagnostic.
- Log fx graph and onnx graph under DEBUG verbosity level for 'fx-graph-to-onnx' diagnostic.
- Adjust unittest to run with diagnostics verbosity level logging.DEBUG.
- Sarif logs will be saved for unittest when `TORCH_LOGS="onnx_diagnostics"` is set.

image

### CODE REVIEW

**Review by justinchuby — APPROVED** [review]

File: torch/onnx/_internal/fx/fx_onnx_interpreter.py [hunk 1]
```diff
@@ -520,6 +541,9 @@ def run(
                     fx_name_to_onnxscript_value,
                 )
 
+        with diagnostic.log_section(logging.DEBUG, "ONNX Graph:"):
+            diagnostic.debug("```\n%s\n```", onnxscript_graph.torch_graph)
```
Comment (titaiwangms) [inline torch/onnx/_internal/fx/fx_onnx_interpreter.py:545]: What's the difference to make this one not LazyString?

**Review by titaiwangms — APPROVED** [review]
Not sure how to use the new api. Is there a page that I can learn?

File: torch/onnx/_internal/fx/fx_onnx_interpreter.py (hunk 1 above)
Comment (BowenBao) [reply torch/onnx/_internal/fx/fx_onnx_interpreter.py:545]: logging api, adopted from python `logging`, lazy formats the string. Hence in this case as long as getattr `onnxscript_graph.torch_graph` is cheap it is fine. The real formatting happens later.

On the contrary. If let's say your argument is `graph_module.print_readable(print=False)`, which runs codegen when called, you'd need to put it behind `LazyString`.

**Comment by BowenBao** [comment]
> Not sure how to use the new api. Is there a page that I can learn?

The docstrings and notes.

**Pull request closed without merging**